src/features/sarcophagi/components/Details.tsx

```diff
@@ -10,32 +11,37 @@ import { DetailsCollapse } from './DetailsCollapse';
 
 export function Details() {
   const { id } = useParams();
-  const { sarcophagus } = useGetSarcophagusDetails({ sarcoId: id });
+  const { sarcophagus } = useGetSarcophagus(id);
   const { address } = useAccount();
   const { bury, isLoading, isBurying } = useBurySarcophagus(id || ethers.constants.HashZero);
   const resurrectionString = buildResurrectionDateString(
     sarcophagus?.resurrectionTime || BigNumber.from(0)
   );
 
+  const { clean, isCleaning } = useCleanSarcophagus(sarcophagus?.id ?? '', address);
+
   // Determine if the rewrap and bury functions are available
   const canRewrapOrBury =
-    sarcophagus?.state === SarcophagusState.Active && sarcophagus?.embalmer === address;
+    sarcophagus?.state === SarcophagusState.Active && sarcophagus?.embalmerAddress === address;
 
   // Determine if the claim function is available
   const canClaim =
-    sarcophagus?.state === SarcophagusState.Resurrected &&
-    sarcophagus?.recipientAddress === address;
+    sarcophagus?.state === SarcophagusState.Resurrected && sarcophagus.recipientAddress === address;
```

Comment:
@DarksightKellar -- does this `Resurrected` state exist anymore? I thought these states got removed from `getSarcophagus`